cmd/lib/healthcheck/oneof.go

```diff
@@ -0,0 +1,55 @@
+package healthcheck
+
+import (
+	"flag"
+	"fmt"
+	"sort"
+	"strings"
+)
+
+type oneof struct {
+	choices map[string]interface{}
+	value   string
+}
+
+var _ flag.Getter = (*oneof)(nil)
+
+func (o *oneof) String() string {
+	return o.value
+}
+
+func (o *oneof) Get() interface{} {
+	return o
+}
+
+func (o *oneof) Set(v string) error {
+	if _, ok := o.choices[v]; ok {
+		o.value = v
+		return nil
+	}
+	return fmt.Errorf("%q is not one of the choices of %s", v, o.printChoices())
+}
+
+func (o oneof) printChoices() string {
```

Comment:
🔕 minor nit, this doesn't actually _print_ anything, maybe something like `choicesStr` would be a more appropriate name?